src/expressions/block-expr.md

```diff
@@ -80,6 +80,64 @@ fn move_by_block_expression() {
 }
 ```
 
+## `async` blocks
+
+> **<sup>Syntax</sup>**\
+> _AsyncBlockExpression_ :\
+> &nbsp;&nbsp; `async` `move`<sup>?</sup> _BlockExpression_
+
+An *async block* is a variant of a block expression which evalutes to
+a *future*. The final expression of the block (if present) determines
+the result value of the future.
+
+Executing an async block is thus similar to executing a closure
+expression: its immediate effect is to produce and return a struct.
+Whereas closures return a struct that implements one or more of the
+`std::ops::Fn` traits, however, the type returned for an async block
+implements the `std::future::Future` trait. The actual data format for
+this type is undefined, but it is roughly equivalent to an enum with
```

Comment:
The second clause (after the "but") should be in a Note also stating that it is what rustc does.